### PR TITLE
React: Regular vs Deployment top up

### PR DESCRIPTION
**Pull request by rzadp**
The TopUp component should not have gas details and `Send 0.005 ETH ...` message when used outside of deployment

### CODE REVIEW

File: universal-login-react/src/App.tsx [hunk 1]
```diff
@@ -142,14 +142,19 @@ export const App = () => {
...
+                  sdk,
+                  isDeployment: false
                 };
                 return (
                   <>
                     <ReactModalContext.Provider value={modalService}>
                       <button id="show-topup-button" onClick={() => modalService.showModal('topUpAccount', topUpProps)}>Show Topup</button>
                       <Modals />
                     </ReactModalContext.Provider>
+                    <ReactModalContext.Provider value={modalService}>
```
Comment (Marik-D) [inline universal-login-react/src/App.tsx]: What's the reason for adding a second modal provider and a second `Modals` component?

Comment (rzadp) [reply universal-login-react/src/App.tsx]: I see your point, updated.

File: universal-login-react/src/core/models/ReactModalContext.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export type TopUpProps = {
   sdk: UniversalLoginSDK
   contractAddress: string;
   onGasParametersChanged: OnGasParametersChanged;
+  isDeployment: boolean;
```
[the remark this answers is not in the document]
Comment (Marik-D) [reply universal-login-react/src/core/models/ReactModalContext.ts]: +1

[the remark this answers is not in the document]
Comment (rzadp) [reply universal-login-react/src/core/models/ReactModalContext.ts]: Good point, done

**Comment by rzadp** [comment]
> I would suggest adding optional children parameter to TopUp component. During deployment, you can show TopUp with GasPrice component as children and in "normal" topUp you can show TopUp component without children.

Created trello story for TopUp component refactoring

File: universal-login-react/src/ui/TopUp/TopUp.tsx [hunk 3]
```diff
@@ -29,6 +30,9 @@ interface TopUpProps {
 }
 
 export const TopUp = ({sdk, onGasParametersChanged, contractAddress, startModal, modalClassName, hideModal, isModal, isDeployment, topUpClassName, logoColor}: TopUpProps) => {
+  if (isDeployment && onGasParametersChanged === undefined) {
```
Comment (xitronix) [inline universal-login-react/src/ui/TopUp/TopUp.tsx]:
```ts
if (isDeployment) {
   ensureNotNull(onGasParametersChanged, MissingParameter, 'onGasParametersChanged')
}
```

Comment (rzadp) [reply universal-login-react/src/ui/TopUp/TopUp.tsx]: 👍 Done.

File: universal-login-react/src/App.tsx [hunk 4]
```diff
@@ -141,13 +141,14 @@ export const App = () => {
                 }
                 const topUpProps: TopUpProps = {
                   contractAddress: Wallet.createRandom().address,
-                  onGasParametersChanged: console.log,
-                  sdk
+                  sdk,
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/App.tsx]: `topUpRegularProps` and `topUpDeploymentProps`

Comment (rzadp) [reply universal-login-react/src/App.tsx]: Done.

**Comment by rzadp** [comment]
Regular top up:

Screenshot 2019-10-17 at 12 03 21
Screenshot 2019-10-17 at 12 03 25

**Comment by rzadp** [comment]
Deployment top up:

Screenshot 2019-10-17 at 12 03 44

**Comment by rzadp** [comment]
Deployment top up:

Screenshot 2019-10-17 at 12 03 44
Screenshot 2019-10-17 at 12 03 47

**Pull request merged**